### PR TITLE
feat: get terminal params from args 

### PR DESCRIPTION
**Pull request by CeerDecy**
#### What this PR does / why we need it:
1. get terminal params from args 
2. more logs in `s_instance_info` watcher

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=628274&iterationID=12783&type=TASK)

#### Specified Reviewers:

/assign @iutx 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |       get terminal params from args and more logs in `s_instance_info` watcher        |
| 🇨🇳 中文    |       terminal参数从args中获取，并改造watcher，多输出些日志       |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by CeerDecy** [comment]
the `Terminal` fun difficult to monkey patch, there are too many `http.Request`'s functions

**Review by iutx — APPROVED** [review]
LGTM @sfwn 
New feature need frontend added params.

File: internal/tools/orchestrator/scheduler/executor/plugins/k8s/instanceinfosync/pod.go [hunk 1]
```diff
@@ -498,6 +500,7 @@ func updatePodAndInstance(dbclient *instanceinfo.Client, podlist *corev1.PodList
 				}
 			}
 		} else {
+			logrus.Infof("get [currentContainerID] from mainContainer,")
```
Comment (sfwn) [inline internal/tools/orchestrator/scheduler/executor/plugins/k8s/instanceinfosync/pod.go:503]: remove the ending comma

File: internal/tools/orchestrator/scheduler/executor/plugins/k8s/instanceinfosync/pod.go [hunk 2]
```diff
@@ -609,7 +612,10 @@ func updatePodAndInstance(dbclient *instanceinfo.Client, podlist *corev1.PodList
 						return orgs, err
 					}
 				}
+				logrus.Infof("lastest instance id %v , others cleared by [prevContainerID]", instances[0].ID)
```
Comment (sfwn) [inline internal/tools/orchestrator/scheduler/executor/plugins/k8s/instanceinfosync/pod.go:615]: maybe `latest`

File: internal/tools/orchestrator/scheduler/executor/plugins/k8s/instanceinfosync/pod.go [hunk 3]
```diff
@@ -683,7 +689,10 @@ func updatePodAndInstance(dbclient *instanceinfo.Client, podlist *corev1.PodList
 						return orgs, err
 					}
 				}
+				logrus.Infof("lastest instance id %v, others cleared by [currentContainerID]", instances[0].ID)
```
Comment (sfwn) [inline internal/tools/orchestrator/scheduler/executor/plugins/k8s/instanceinfosync/pod.go:692]: same

**Review by sfwn — CHANGES_REQUESTED** [review]
May need some unit tests :-)

File: internal/tools/orchestrator/endpoints/instance.go [hunk 4]
```diff
@@ -315,9 +315,18 @@ func (e *Endpoints) getPodStatusFromK8s(runtimeID, serviceName string) ([]apistr
...
+		// TODO change `DICE_CLUSTER_NAME` to  `core.erda.cloud/cluster-name`
 		clusterName := ""
-		if _, ok := pod.Labels["DICE_CLUSTER_NAME"]; ok {
-			clusterName = pod.Labels["DICE_CLUSTER_NAME"]
+		for _, container := range pod.Spec.Containers {
+			if clusterName != "" {
+				break
+			}
+			for _, env := range container.Env {
+				if env.Name == "DICE_CLUSTER_NAME" {
```
Comment (sfwn) [inline internal/tools/orchestrator/endpoints/instance.go:325]: you'd better use `apistructs.DICE_CLUSTER_NAME`.
literal string is not recommended.

File: internal/tools/orchestrator/scheduler/impl/terminal/terminal.go [hunk 5]
```diff
@@ -126,88 +129,109 @@ func Terminal(clusterSvc clusterpb.ClusterServiceServer, w http.ResponseWriter,
...
-	instance := instances[0]
+	clustername = r.URL.Query().Get("clusterName")
 
-	// 3. Check permissions
-	access := false
-	if instance.OrgID != "" {
-		orgid, err := strconv.ParseUint(instance.OrgID, 10, 64)
+	if args.PodName != "" && args.Namespace != "" && args.ContainerName != "" {
+		logrus.Infof("get termial connect param from args")
+		orgID, _ = strconv.Atoi(r.Header.Get("org-id"))
```
Comment (sfwn) [inline internal/tools/orchestrator/scheduler/impl/terminal/terminal.go:150]: httputil.OrgHeader

File: internal/tools/orchestrator/scheduler/impl/terminal/terminal.go [hunk 6]
```diff
@@ -126,88 +129,109 @@ func Terminal(clusterSvc clusterpb.ClusterServiceServer, w http.ResponseWriter,
...
+		// 3. Check permissions
+		access := false
+		if instance.OrgID != "" {
+			orgid, err := strconv.ParseUint(instance.OrgID, 10, 64)
+			if err != nil {
+				logrus.Errorf("failed to parse orgid for instance: %v, %v", instance.ContainerID, err)
+				return
+			}
+			p, err := bundle.New(bundle.WithErdaServer()).CheckPermission(&apistructs.PermissionCheckRequest{
+				UserID:   r.Header.Get("User-ID"),
```
Comment (sfwn) [inline internal/tools/orchestrator/scheduler/impl/terminal/terminal.go:182]: httputil.userheader

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**